pandas/core/categorical.py

```diff
@@ -308,6 +308,9 @@ def __init__(self, values, categories=None, ordered=None, dtype=None,
         elif isinstance(values, (ABCIndexClass, ABCSeries)):
             # we'll do inference later
             pass
+        elif getattr(values, 'ndim', 0) > 1:
+            raise TypeError("> 1 ndim Categorical are not "
```

Comment:
Should this be a `NotImplementedError` instead of a `TypeError`?  Similar to how the existing error with this message on line 350 is a `NotImplementedError`?

Reply:
Good call.